### PR TITLE
[SPARK-38239][PYTHON][MLLIB] Fix pyspark.mllib.LogisticRegressionModel.__repr__

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

Adds native implementation of `__repr__` to `pyspark.mllib.classification.LogisticRegressionModel`.  [Documentation example](https://spark.apache.org/docs/latest/mllib-linear-methods.html#logistic-regression) returns the following after this patch:

```python
>>> model
pyspark.mllib.LogisticRegressionModel: intercept = 0.0, numFeatures = 16, numClasses = 2, threshold = 0.5

```

### Why are the changes needed?

Current implementation is copied from `pyspark.ml.classification` counterpart and fails with `AttributeError`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual testing.

### CODE REVIEW

File: python/pyspark/mllib/classification.py [hunk 1]
```diff
@@ -278,7 +278,10 @@ def load(cls, sc, path):
         return model
 
     def __repr__(self):
```
Comment (zero323) [inline python/pyspark/mllib/classification.py:280]: Strictly speaking, this should be `__str__` not `__repr__`.

**Comment by HyukjinKwon** [comment]
Looks making sense .. but @WeichenXu123 would you mind taking a look please? I don't have enough background on string representation of ML instances.

**Comment by HyukjinKwon** [comment]
@zero323, quick question

> Current implementation is copied from `pyspark.ml.classification` counterpart and fails with `AttributeError`.

Do you know why it fails?

**Comment by zero323** [comment]
@HyukjinKwon  There is simply no such method define. `_call_java` is part of the `pyspark.ml.wrapper.JavaWrapper` API, which is not used in `pyspark.mllib`. Also, JVM model is not captured in this (and most of the `pyspark.mllib` if I recall correctly), so there is no object on which such method could be invoked ‒ if there was, we'd use something like tree models:

https://github.com/apache/spark/blob/3a750ca4686e85e715c48c688f48acf7851144ed/python/pyspark/mllib/tree.py#L76-L78

In general, `mllib` models don't have consistent string representation ‒ some just use `object` implementation, some summary like this one, some show [coefficients](https://github.com/apache/spark/blob/3a750ca4686e85e715c48c688f48acf7851144ed/python/pyspark/mllib/regression.py#L110-L111), and every time we use `__repr__` instead of `__str__`. If API wasn't in maintenance mode, that be something to address I guess...

**Comment by srowen** [comment]
Are there others that need fixing too? or at least a subset that could use the same fix?

**Comment by zero323** [comment]
> Are there others that need fixing too? or at least a subset that could use the same fix?

Nope. This one is just one time copy-and-paste from `pyspark.ml`. Other models either have no `__repr__` or one that is part of the original implementation.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**